s3torchconnector/src/s3torchconnector/lightning/s3_lightning_checkpoint.py

```diff
@@ -0,0 +1,82 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+
+from typing import Optional, Dict, Any
+
+import torch
+
+from lightning.pytorch.plugins.io import CheckpointIO
+
+from .._s3client import S3Client
+from .._s3dataset_common import parse_s3_uri
+
+
+class S3LightningCheckpoint(CheckpointIO):
+    """A checkpoint manager for S3 using the :class:`CheckpointIO` interface."""
+
+    def __init__(self, region: str):
+        self.region = region
+        self._client = S3Client(region)
+
+    def save_checkpoint(
+        self,
+        checkpoint: Dict[str, Any],
+        path: str,
+        storage_options: Optional[Any] = None,
+    ) -> None:
+        """Save model/training states as a checkpoint file through state-dump and upload to S3.
+
+        Args:
+            checkpoint: dict containing model and trainer state
+            path: write-target path
+            storage_options: Optional parameters when saving the model/training states.
+
+        """
+        self._validate_path(path)
+        bucket, key = parse_s3_uri(path)
+        with self._client.put_object(bucket, key) as s3writer:
+            torch.save(checkpoint, s3writer)
+
+    def load_checkpoint(
+        self,
+        path: str,
+        map_location: Optional[Any] = None,
+    ) -> Dict[str, Any]:
+        """Load checkpoint from a path when resuming or loading ckpt for test/validate/predict stages.
```

Comment:
`from an S3 location` perhaps?
I would avoid path since we are supporting valid S3 URIs in string format.